x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/analytics/DataFrameDataExtractorFactory.java

```diff
@@ -33,6 +37,12 @@
     private static final List<String> IGNORE_FIELDS = Arrays.asList("_id", "_field_names", "_index", "_parent", "_routing", "_seq_no",
         "_source", "_type", "_uid", "_version", "_feature", "_ignored");
 
+    /**
+     * The types supported by data frames
+     */
+    private static final Set<String> COMPATIBLE_FIELD_TYPES = new HashSet<>(Arrays.asList(
```

Comment:
If you want to support all numeric types, I would suggest doing something similar to what Rollups does:
```suggestion
    private static final Set<String> COMPATIBLE_FIELD_TYPES;
    static {
        Set<String> types = Stream.of(NumberFieldMapper.NumberType.values())
            .map(NumberFieldMapper.NumberType::typeName)
            .collect(Collectors.toSet());
        types.add("scaled_float"); // have to add manually since scaled_float is in a module
        COMPATIBLE_FIELD_TYPES = types;
    }
```

Reply:
Great point!